File: go/arrays/valid_anagram_test.go
package arrays

import "testing"

func TestValidAnagram(t *testing.T) {
	tests := []struct {
		name     string
		s, t     string
		expected bool
	}{
		{"valid anagram", "anagram", "nagaram", true},
		{"not anagram", "rat", "car", false},
		{"empty strings", "", "", true},
		{"different lengths", "a", "ab", false},
		{"duplicate chars", "aab", "baa", true},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			result := ValidAnagram(tt.s, tt.t)
			if result != tt.expected {
				t.Errorf("got %v, want %v", result, tt.expected)
			}
		})
	}
}

File: typescript/arrays/validAnagram.test.ts
import { validAnagram } from './validAnagram';

describe('validAnagram', () => {
  it('should return true for valid anagram', () => {
    expect(validAnagram('anagram', 'nagaram')).toBe(true);
  });

  it('should return false for non-anagram', () => {
    expect(validAnagram('rat', 'car')).toBe(false);
  });

  it('should return true for empty strings', () => {
    expect(validAnagram('', '')).toBe(true);
  });

  it('should return false for different lengths', () => {
    expect(validAnagram('a', 'ab')).toBe(false);
  });

  it('should handle duplicate characters', () => {
    expect(validAnagram('aab', 'baa')).toBe(true);
  });
});
